### DISCUSSION TITLE
Error parsing big files with negative coordinates (concatenated field)

### DISCUSSION THREAD

**Discussion by tubiana** [Electrostatics/apbs#221] [General] [answered]
Hi ! 

I'm trying to compute an APBS map on a rather big system that contains negatives coordinates under 100A. 
Example ; 
```
ATOM  52291  O   ASP O  58     -45.751-100.406  19.252 -0.5819 1.6612
```
Since there is no space between coord X and coord Y, APBS return me this error : 
```
Valist_readPQR:  Error parsing atom 52291!
Please double check this atom in the pqr file, e.g., make sure there are no concatenated fields.
```
\
Is APBS read the pqr file and use a space as separator ? What would be the best workaround according to you ?
I could remove the last decimal of the atom but I just wanted to be sure if you have any better solution :-) 

Best regards,
Thibault Tubiana.

I don't know if it is related but i'm on Mac with the last Darwin build.

**Comment by sobolevnrm** [comment, marked as answer]
Hello.  Which version of APBS are you using?

You can pre-process the file using [PDB2PQR](http://www.poissonboltzmann.org/) with the `--whitespace` option to add spaces between the columns.

**Comment by tubiana** [comment]
I use the version 3.4.0 but I just tested with a pqr prepared with `--whitespace` and it works perfectly ! 
Thank you so much  😃

**Comment by sobolevnrm** [comment]
Good news; thank you!

**Reply by spmorton** [reply]
This is a pretty cheezy solution, why not properly parse the valid file? 
PDB2PQR reads/writes it just fine per PDB spec but APBS cannot?

**Reply by sobolevnrm** [reply]
We needed to be able to handle the more general case when coordinates were too large for the official spec.

More generally, mmCIF is a better long-term solution.

**Comment by spmorton** [comment]
That all may be true, however, for those third party apps like VMD that are expecting the standards based layout, this does not allow them to properly read the PQR file until the non-uniform layout can be handled.
